crates/types/src/config/common.rs

```diff
@@ -385,11 +436,29 @@ impl Default for CommonOptions {
                 Some(15),
                 Some(Duration::from_secs(5)),
             ),
-            auto_provision_partitions: true,
+            cluster_configuration: ClusterConfigurationType::default(),
+            default_cluster_configuration: None,
         }
     }
 }
 
+/// # ClusterConfigurationType
+///
+/// Configure initial cluster configuration.
+#[serde_as]
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+#[serde(rename_all = "kebab-case")]
+pub enum ClusterConfigurationType {
+    #[default]
+    /// Use sane default cluster configuration mainly
+    /// suitable for a single node setup
+    Default,
```

Comment:
Not really. I think the docstr might be misleading here. Since Default still respects the bifrost `default-provider`. So it's not necessary for a single node.

I will improve on the doc